pkg/apis/ceph.rook.io/v1/cluster.go

```diff
@@ -81,8 +81,8 @@ func validateUpdatedCephCluster(updatedCephCluster *CephCluster, found *CephClus
 		return errors.Errorf("invalid update: Provider change from %q to %q is not allowed", found.Spec.Network.Provider, updatedCephCluster.Spec.Network.Provider)
 	}
 
-	for i, storageClassDeviceSet := range updatedCephCluster.Spec.Storage.StorageClassDeviceSets {
-		if storageClassDeviceSet.Encrypted != found.Spec.Storage.StorageClassDeviceSets[i].Encrypted {
+	for i, storageClassDeviceSet := range found.Spec.Storage.StorageClassDeviceSets {
+		if storageClassDeviceSet.Encrypted != updatedCephCluster.Spec.Storage.StorageClassDeviceSets[i].Encrypted {
```

Comment:
Either way, we can't use `i` from a different cluster spec or else we could hit an exception if they are adding/removing device sets. Perhaps we should just perform this check if the length of the storageClassDeviceSets is the same in the two cluster specs.

Reply:
I see, adding will not cause issue but removing will cause issue.

Reply:
made the changes but will have problem in case we change the encryption type and remove/add storage sets, the webhook will not able to validate that since the count is not same.

Reply:
Agreed, we will miss validation in that case. I'm thinking the complexity of handling that case will just not be worth it.

Reply:
As I already [wrote in the original issue](https://github.com/rook/rook/issues/10222#issuecomment-1119763160), it would be worth comparing elements by name instead of indexes. in this case, it does not matter in what order and how many elements are in the list.
  Shouldn't names be unique? If there is no such check, then it would be worth adding it too.
  
@travisn , think about it in the complex, maybe it's worth getting a separate issue.

Reply:
Yes, comparing names would be better of course. It's not really that complex so we could implement it in a follow-up PR, but with this PR at least the crash is avoided. To compare by name we would just have to build a map to facilitate looking up the names to avoid the n^2 double-for loop.